modules/api/php/endpoints/candidate/visit/visit.class.inc

```diff
@@ -95,23 +108,19 @@ class Visit extends Endpoint implements \LORIS\Middleware\ETagCalculator
             if (!$this->_visit->isAccessibleBy($user)) {
                 return new \LORIS\Http\Response\JSON\Forbidden('Permission denied');
             }
-
         }
 
         $pathparts = $request->getAttribute('pathparts');
         if (count($pathparts) === 0) {
-            switch ($request->getMethod()) {
-            case 'GET':
-                return $this->_handleGET();
+            $method = $request->getMethod();
 
-            case 'PUT':
-                return $this->_handlePUT($request);
-
-            case 'OPTIONS':
+            if (in_array($method, $this->allowedMethods())) {
+                $handle = 'handle'.$method;
+                return $this->$handle($request);
+            } else if ($method === 'OPTIONS') {
```

Comment:
There is not `else` in the API. (with 1 exception that I never took the time to fix)